Feature/cax portal@axios instance

fatihayyildiz opened the pull request:
Generic axios instance created
Partnernetwork axios usage implemented
PartnerNetwork routing implemented
Partner network api response implemented at Redux store
Alternative usage of Redux store implemented

fatihayyildiz committed:
WIP - Some progress about axios instance usage

fatihayyildiz committed:
BpdmSlice redux store implemented
slice component usage implemented

fatihayyildiz committed:
Bpdm wording changed to partner network
Renaming of components and store values
PartnerNetwork routing implemented

oyo commented on portal/code/cx-portal/.env:
```diff
@@ -1,2 +1,4 @@
 REACT_APP_BASE_CENTRAL_IDP=https://catenaxdev003akssrv.germanywestcentral.cloudapp.azure.com/iamcentralidp/auth
 REACT_APP_BASE_UTIL=https://catenax-dev003-util.azurewebsites.net/v1
+REACT_APP_BPDM_API_BASE_URL=https://catenax-bpdm-dev.germanywestcentral.cloudapp.azure.com/api
```
calling any endpoint of this api results in a timeout

fatihayyildiz replied:
I want to work on Error handling may be in another PR because it looks like quite a lot to do there. I agree it's need definitely needs more proper error handling in api calls. For timeout issue, I also get timeout. Most probably that api is down for a moment. I will inform my responsible colleague to fix it. Swagger ui url : https://catenax-bpdm-dev.germanywestcentral.cloudapp.azure.com/api/swagger-ui

oyo replied:
We should not expose buggy features to the test users because this leaves a bad impression. So change the access to this page from EVERYONE to some role that only we developers or people who want to test that feature can see. So in Keycloak create a role portal_developer or partner_network_developer that you assign to yourself: 
`AccessService.ts` line 17
`[PAGES.PARTNER_NETWORK]: ROLES.PORTAL_DEVELOPER`
Once the feature is working without errors we can open it to the correct target audience

fatihayyildiz replied:
Totally agree, I will update the PR regarding this point

oyo commented on portal/code/cx-portal/src/state/api/partnerNetwork/partnerNetworkApi.ts:
```diff
@@ -0,0 +1,25 @@
+
+  // TODO: Token needs to read from Redux store
+  public constructor(token: string) {
+    super(`${process.env.REACT_APP_BPDM_API_BASE_URL}`,{'Authorization': `Bearer ${token}`})
```
why not simply
`super(process.env.REACT_APP_BPDM_API_BASE_URL, `

fatihayyildiz replied:
I already mention the issue at the comment of Timeout value. Typescript didnt consider this variable as a string, I need to cast it somehow :)

oyo commented on portal/code/cx-portal/src/utils/HttpClient.ts:
```diff
@@ -0,0 +1,44 @@
+                        headers: AxiosRequestHeaders = {
+                          'Content-Type': 'application/json'
+                        },
+                        timeout: number = Number.parseInt(`${process.env.REACT_APP_API_CALL_TIMEOUT}`)) {
```
another
`Number.parseInt(process.env.REACT_APP_API_CALL_TIMEOUT)`

fatihayyildiz replied:
Typescript didnt understand this env parameter as a string. It should cast to string type like this way or may be like this way Number.parseInt(process.env.REACT_APP_API_CALL_TIMEOUT as string). using this back-ticks looks simpler to me

oyo reviewed (changes requested):
API call is always running into a timeout - please add proper error handling or describe the steps to get this running.

fatihayyildiz committed:
Bpdm wording for types changed to PartnerNetwork
PartnerNetwork Redux initial state type created
Rootstate created for type of store values
createSelector is removed

oyo reviewed (approved):
I approve this for now so we can continue - we can fix the role issue in a separate PR
